apps/council-frontend/src/ui/delegate/DelegatesList/DelegatesList.tsx

```diff
@@ -33,6 +41,17 @@ function DelegatesList({
   selectedDelegate,
   setDelegateAddressInput,
 }: DelegatesListProps): ReactElement {
+  const delegateSearchBarInputRef = useRef<string>("");
+  const updateInputRef = (value: string) => {
+    delegateSearchBarInputRef.current = value;
+  };
+
+  const [delegateListFilter, setDelegateListFilter] = useState("");
+
+  const handleDelegateListFiltering = () => {
+    setDelegateListFilter(delegateSearchBarInputRef.current.toLowerCase());
+  };
```

Comment:
Instead of maintaining the value on each change with the ref, we can just include the input value in the `onFiltering` callback and set the state directly here.